File: Leetcode/LongestPalindromicSubstring.cpp
class Solution {
public:
    string longestPalindrome(string& s) {
        int n = s.size(); 
        int ans = 1;
        int start = 0; 
        
        for(int i = 0;i < n-1; ++i){
            // odd string
            //s[i] is the middle 
            
            int low = i-1, high = i+1; 
            
            while(low >= 0 && high < n && s[low] == s[high]) low--, high++; 
            
            int len = (high - low + 1) - 2; 
            
            if(len > ans){
                start = low+1; 
                ans = len; 
            }
            
            // even string
            low = i; high = i+1; 
                
            while(low >= 0 && high < n && s[low] == s[high]) low--, high++; 
            
            len = (high-low+1) - 2; 
            
            if(len > ans){
                start = low+1; 
                ans = len; 
            }
        }
        
        return s.substr(start, ans); 
    }
};

File: Leetcode/BlindLeetCode 76 Problems/LongestPalindromicSubstring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        n = len(s)
        maxLength = 1
        start = 0
        
        for i in range(n-1) : 
            if i+1 < n and s[i] == s[i+1] : 
                j,k = i-1, i+2
                
                while j >= 0 and k < n and s[j] == s[k]: 
                    j -= 1;
                    k += 1; 
                
                j += 1
                k -= 1 
                
                if maxLength < k-j+1 : 
                    maxLength = k-j+1; 
                    start = j; 
            
            j,k = i-1,i+1
            
            while j >= 0 and k < n and s[j] == s[k]: 
                j -= 1
                k += 1
            
            j += 1
            k -= 1
            
            if maxLength < k-j+1 : 
                maxLength = k-j+1
                start = j
                
        return s[start:start+maxLength]